spec/models/post_spec.rb

```diff
@@ -0,0 +1,98 @@
+  fourth_comment = Comment.new(post: first_post, user: second_user, text: 'Hi Tom!')
+  fifth_comment = Comment.new(post: first_post, user: second_user, text: 'Hi Tom!')
+  sixth_comment = Comment.new(post: first_post, user: second_user, text: 'Hi Tom!')
+
+  context 'Write validation tests for Post Model' do
+    it 'is not valid without a title' do
+      first_post.title = nil
+      expect(first_post).to_not be_valid
+    end
+
+    it 'is not valid if title exceed 250 characters' do
+      first_post.title = 'M' * 251
+      expect(first_post).to_not be_valid
+    end
+
+    it 'is valid with a name' do
+      first_post.text = 'This is the content of the post.'
+      expect(first_post).to_not be_valid
+    end
+
+    it 'is not valid if comments_counter is not integer' do
+      first_post.comments_counter = 'Mavericks'
+      expect(first_post).to_not be_valid
+    end
+
+    it 'is not valid if comments_counter is negative' do
+      first_post.comments_counter = -5
+      expect(first_post).to_not be_valid
+    end
+
+    it 'is not valid if likes_counter is not an integer' do
+      first_post.likes_counter = 'Balitaan'
+      expect(first_post).to_not be_valid
+    end
+
+    it 'is not valid if likes_counter is negative' do
+      first_post.likes_counter = -21
+      expect(first_post).to_not be_valid
+    end
+  end
+
+  context 'Write unit tests for Post Model Methods' do
+    it 'returns 0 for no comment' do
+      number_of_comments = first_post.recent_5_comments.count
+      expect(number_of_comments).to be 0
+    end
+
+    it 'returns 1 for one comment' do
+      first_comment.save
+      number_of_comments = first_post.recent_5_comments.count
+      expect(number_of_comments).to be 1
+    end
+
+    it 'returns 5 for five comments' do
+      second_comment.save
+      third_comment.save
+      fourth_comment.save
+      fifth_comment.save
+      sixth_comment.save
+
+      comments = first_post.recent_5_comments
+      number_of_comments = first_post.recent_5_comments.count
+      texts = comments.pluck(:text)
+      expect(number_of_comments).to be 5
+      expect(texts).to eql [sixth_comment.text, fifth_comment.text, fourth_comment.text, third_comment.text,
+                            second_comment.text]
+    end
+  end
+
+  context 'Write unit tests for Post Model Methods' do
+    it 'updates post counter equal to 1 for a given user' do
+      first_post.update_posts_counter
+      number_of_posts = second_user.posts_counter
+      expect(number_of_posts).to be 1
+    end
+
+    it 'updates post counter equal to 2 for a given user' do
+      first_post.update_posts_counter
+      second_post = Post.create(user: second_user, title: 'Hello', text: 'This is my second post', comments_counter: 2,
+                                likes_counter: 3)
+      second_post.update_posts_counter
+      number_of_posts = second_user.posts_counter
+      expect(number_of_posts).to be 2
+    end
+  end
```

Comment:
- [ ] Also, here in the posts_spec, you are expected to write tests for the validations. Please add them. I have given you the example of how to do it in the above comments 😃 👍